pkg/apis/v1alpha1/staticroute_types.go

```diff
@@ -0,0 +1,79 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type StaticRouteStatusCondition string
+
+const (
+	StaticRouteReady    StaticRouteStatusCondition = "Ready"
+	StaticRouteNotReady StaticRouteStatusCondition = "NotReady"
+)
+
+// StaticRouteCondition defines condition of StaticRoute.
+type StaticRouteCondition struct {
+	// Type defines condition type.
+	Type StaticRouteStatusCondition `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about condition.
+	Message string `json:"message,omitempty"`
+}
+
+// StaticRouteSpec defines static routes configuration on VPC
+type StaticRouteSpec struct {
+	// Specify network address in CIDR format
+	// +kubebuilder:validation:Format=cidr
+	Network string `json:"network"`
+	// Next hop gateway
+	NextHops []NextHop `json:"next_hops"`
+}
+
+//  NextHop defines next hop configuration for network
+type NextHop struct {
+	// Next hop gateway IP address
+	// +kubebuilder:validation:Format=ip
+	IpAddress string `json:"ip-address"`
+	// Cost associated with next hop route
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=255
+	AdministrativeDistance int `json:"administrative_distance,omitempty"`
```

Comment:
in phase-1, we only have 1 next hop with node ip as ip_address, the admin_distance would be all 1. So I think this field can be removed.